mmdet/core/hook/neptunelogger_hook.py

```diff
@@ -0,0 +1,305 @@
+            Requires EvalHook.
+            Default: 50
+    """
+
+    INTEGRATION_VERSION_KEY = 'source_code/integrations/neptune-mmdetection'
+
+    def __init__(self,
+                 *,
+                 init_kwargs: Optional[Dict] = None,
+                 api_token: str = None,
+                 project: str = None,
+                 interval: int = 10,
+                 base_namespace: str = 'training',
+                 log_model: bool = False,
+                 log_checkpoint: bool = False,
+                 num_eval_predictions: int = 5,
+                 **kwargs) -> None:
+        super().__init__(interval=interval, **kwargs)
+
+        if init_kwargs:
+            api_token = init_kwargs.pop('api_token', None) or api_token
+            project = init_kwargs.pop('project', None) or project
+
+        self._run = neptune.init_run(
+            api_token=api_token,
+            project=project,
+            **init_kwargs if init_kwargs else {})
+        self.base_namespace = base_namespace
+        self.base_handler = self._run[base_namespace]
+
+        self.log_model = log_model
+        self.log_checkpoint = log_checkpoint
+
+        self.num_eval_predictions = num_eval_predictions
+        self.log_eval_predictions = (num_eval_predictions > 0)
+
+        self.ckpt_hook: CheckpointHook | None = None
+        self.ckpt_interval: int | None = None
+
+        self.eval_hook: EvalHook | None = None
+        self.eval_interval: int | None = None
+
+        self.val_dataset = None
+
+    def _log_integration_version(self) -> None:
+        self._run[self.INTEGRATION_VERSION_KEY] = version.__version__
+
+    def _log_config(self, runner) -> None:
+        if runner.meta is not None and runner.meta.get('exp_name',
+                                                       None) is not None:
+            src_cfg_path = osp.join(runner.work_dir,
+                                    runner.meta.get('exp_name', None))
+            config = Config.fromfile(src_cfg_path)
+            if osp.exists(src_cfg_path):
+                self.base_handler['config'] = config.pretty_text
+        else:
+            runner.logger.warning('No meta information found in the runner. ')
+
+    def _add_ground_truth(self, runner):
+        # Get image loading pipeline
+        from mmdet.datasets.pipelines import LoadImageFromFile
+        img_loader = None
+        for t in self.val_dataset.pipeline.transforms:
+            if isinstance(t, LoadImageFromFile):
+                img_loader = t
+
+        if img_loader is None:
+            self.log_evaluation = False
+            runner.logger.warning(
+                'LoadImageFromFile is required to add images '
+                'to the Neptune run.')
+            return
+
+        # Select the images to be logged.
+        eval_image_indices = np.arange(len(self.val_dataset))
+        # Set seed so that same validation set is logged each time.
+        np.random.seed(42)
```

Comment:
I think this is not good as it will silently set the global seed for Numpy.

We can probably reuse `self.eval_image_indices` so that we log same samples. I think it is ok to log any 50 samples between different runs.

Reply:
How do you mean? This class doesn't have an attribute like that. Plus numpy is listed in global requirements already (build requirements and test requirements I think)

Reply:
More context here : https://builtin.com/data-science/numpy-random-seed

Reply:
Got it. So shall I change it to sth like:
```# Set seed so that same validation set is logged each time.
    rng = np.random.default_rng(42)
    rng.shuffle(eval_image_indices)
    self.eval_image_indices = eval_image_indices[:self.num_eval_predictions]
```
?

Reply:
I think this can we tweaked to the following so that we don't need to recompute the eval_image_indices. Wdyt?

```python
if not self.eval_image_indices:
  rng = np.random.default_rng(42)
  rng.shuffle(eval_image_indices)
  self.eval_image_indices = eval_image_indices[:self.num_eval_predictions]
```